src/citrine/resources/project.py

```diff
@@ -430,6 +432,45 @@ def remove_user(self, user_uid: Union[str, UUID]) -> bool:
         )
         return True
 
+    def gemd_batch_delete(self, id_list: Union[List[Tuple[str, str]], List[UUID]]) -> \
+            List[Tuple[str, str, ApiError]]:
```

Comment:
For all the confusion and inconsistency in these signature, you should probably copy the sig from `data_concepts.delete` to maintain some consistency.  That, unfortunately, would remove support for external IDs, but is futureproof to enabling them later.

Reply:
What do you mean, 'futureproof to enabling them later'. Why would we purposefully exclude the ability to delete via external ID when that works now?
I've looked at the signature for .delete(), and it doesn't seem to really apply for a bulk scenario:
 - You would have to supply a single scope for all IDs, that seems suboptimal
 - We don't support dry_run in a batch delete scenario, and that would be challenging due to the inbound links that are progressively resolved (and depend on the deleted items)
 - .delete() blindly returns a successful Response object, which doesn't provide nearly the detail on failed items that this does

Reply:
This difference of opinion is driven, in part, by the lack of actual customer signal.  To my understanding, the users who have need for this feature at present are 1) DE for cleaning up after a failed load and 2) engineering for speeding up tests.  Neither of those scenarios is substantially benefited by allowing a wider range of IDs.  I was not suggesting that we also include `dry_run` though I understand I was ambiguous on that point.

Right now, call signatures in citrine-python are very heterogenous.  I don't think we have sufficient justification at present to refactor to address this, but this particular signature choice (a scope-UID tuple) is introducing yet another pattern.  

If @djack concurs with your analysis, I'll swap to an approve.

Reply:
Ken's point is not unreasonable - the users of this endpoint are (for now) going to want to be able to copy/paste and have things just work. This signature forces them to think a bit more if they want to make use of this endpoint. 

However, my instinct here is that Dave's signature is pretty natural for a batch endpoint on gemd objects than the `.delete` and `.get` signatures we have today. It's not so much a brand new pattern as it is a version of the same pattern that is list valued, as opposed to individually valued. I do think we could paper over a lot of these issues by (also) accepting a https://github.com/CitrineInformatics/gemd-python/blob/master/gemd/entity/link_by_uid.py in the union type. I have a suspicion that regardless of whether we follow @dleblanc or @kroenlein's path, there are going to be a lot of helper functions that transform / split linkbyuids.

I think any frustrations about API consistency could likely be papered over by being consistent when we add batch-get support to citrine-python https://citrine.atlassian.net/browse/PLA-6574

Reply:
Nice call on LinkByUID as the identifier wrapper.  That also matches well with endpoints that take objects directly.

Reply:
W.r.t. the tuple, I can't find an instance in either gemd-python or citrine-python where we have a scope-uid tuple.  If someone can find an instance of that, I'll approve.  Otherwise, I think we should remove the tuple from the argument list.